### PR TITLE
fix: 1704 Seedlots not updating BV values correctly

### LINKED ISSUE #1704: Seedlots not updating BV values correctly
**Describe the Bug**
After updating the BV values for specific parent trees on old SPAR, the changes are not being applied submitted seedlots on new SPAR, which affects new calculations on the parent tree step.

**Expected Behaviour**
It should load the new BV values on the parent tree table and calculate the new values correctly.

**Actual Behaviour**
BV values that are loaded are not the ones updated on old SPAR, leading to incorrect calculations.

**Additional context**
This task is a continuation of task #1667.

### PR DESCRIPTION
**Pull request by mgaseta**
# Description
Closes #1704

These issues will be better tested on the test environment, due the statuses of the seedlots with issues.

#### New
- Nothing :)

#### Changed
- Changed a validation when populating the table
- Changed a validation when submitting the calculation data

#### Removed
- Nothing :)

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [ ] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-16-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1716-backend.apps.silver.devops.gov.bc.ca/swagger-ui/index.html)
- [Oracle-API](https://nr-spar-1716-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

### CODE REVIEW

**Review by SLDonnelly — COMMENTED** [review]
The recalculation is working as expected in DEV, but I will need to test whether new BVs are being pulled in correctly in SPAR2 TEST

File: frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/utils.ts [hunk 1]
```diff
@@ -741,15 +744,21 @@ export const generatePtValCalcPayload = (
...
   rows.forEach((row) => {
     const newPayloadItem: OrchardParentTreeValsType = {
       parentTreeId: findParentTreeId(state, row.parentTreeNumber.value),
       parentTreeNumber: row.parentTreeNumber.value,
       coneCount: Number(row.coneCount.value),
       pollenCount: Number(row.pollenCount.value),
-      smpSuccessPerc: Number(row.smpSuccessPerc.value),
+      smpSuccessPerc:
+        row.smpSuccessPerc.value === 'null'
+          ? Number(row.smpSuccessPerc.value)
```
Comment (xiaopeng0202) [inline frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/utils.ts:758]: if  row.smpSuccessPerc.value is 'null', Number(row.smpSuccessPerc.value) will be NaN, is that expected?

**Review by xiaopeng0202 — CHANGES_REQUESTED** [review]

File: frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/utils.ts (hunk 1 above)
Comment (mgaseta) [reply frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/utils.ts:756]: Fixed here: https://github.com/bcgov/nr-spar/pull/1716/commits/8b288bd746865cfde323b5020e8a23817e69fae5

File: frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/utils.ts [hunk 2]
```diff
@@ -736,6 +736,9 @@ export const generatePtValCalcPayload = (
   pollenContaminantBreedingValue?: string
 ): PtValsCalcReqPayload => {
   const { tableRowData, mixTabData } = state;
+
+  // eslint-disable-next-line no-debugger
+  debugger;
```
[the remark this answers is not in the document]
Comment (mgaseta) [reply frontend/src/components/SeedlotRegistrationSteps/ParentTreeStep/utils.ts]: My bad... Fixed here: https://github.com/bcgov/nr-spar/pull/1716/commits/8b288bd746865cfde323b5020e8a23817e69fae5

**Review by xiaopeng0202 — APPROVED** [review]

**Pull request merged**